contracts/verifying/VerifyingSingletonPaymasterV2.sol

```diff
@@ -1,9 +1,363 @@
+    uint256 private unaccountedEPGasOverhead;
+
+    mapping(address => uint256) public paymasterIdBalances;
+
+    address public verifyingSigner;
+
+    uint32 public fixedPriceMarkup; // immutable? constant? 1.1e6
+
+    // Review if fixed markup is needed in constructor (can init with value or make constant(and remove setter))
+    constructor(
+        address _owner,
+        IEntryPoint _entryPoint,
+        address _verifyingSigner,
+        uint32 _fixedPriceMarkup
+    ) payable BasePaymaster(_owner, _entryPoint) {
+        if (address(_entryPoint) == address(0)) revert EntryPointCannotBeZero();
+        // Review // maybe <= 1300000
+        require(_fixedPriceMarkup <= PRICE_DENOMINATOR * 2, "markup too high");
+        if (_verifyingSigner == address(0))
+            revert VerifyingSignerCannotBeZero();
+        assembly {
+            sstore(verifyingSigner.slot, _verifyingSigner)
+        }
+        unaccountedEPGasOverhead = 12000;
+        fixedPriceMarkup = _fixedPriceMarkup;
+    }
+
+    /**
+     * @dev Add a deposit for this paymaster and given paymasterId (Dapp Depositor address), used for paying for transaction fees
+     * @param paymasterId dapp identifier for which deposit is being made
+     */
+    function depositFor(address paymasterId) external payable nonReentrant {
+        if (paymasterId == address(0)) revert PaymasterIdCannotBeZero();
+        if (msg.value == 0) revert DepositCanNotBeZero();
+        paymasterIdBalances[paymasterId] =
+            paymasterIdBalances[paymasterId] +
+            msg.value;
+        entryPoint.depositTo{value: msg.value}(address(this));
+        emit GasDeposited(paymasterId, msg.value);
+    }
+
+    /**
+     * @dev get the current deposit for paymasterId (Dapp Depositor address)
+     * @param paymasterId dapp identifier
+     */
+    function getBalance(
+        address paymasterId
+    ) external view returns (uint256 balance) {
+        balance = paymasterIdBalances[paymasterId];
+    }
+
+    /**
+     @dev Override the default implementation.
+     */
+    function deposit() public payable virtual override {
+        revert("user DepositFor instead");
+    }
+
+    /**
+     * @dev Withdraws the specified amount of gas tokens from the paymaster's balance and transfers them to the specified address.
+     * @param withdrawAddress The address to which the gas tokens should be transferred.
+     * @param amount The amount of gas tokens to withdraw.
+     */
+    function withdrawTo(
+        address payable withdrawAddress,
+        uint256 amount
+    ) public override nonReentrant {
+        if (withdrawAddress == address(0)) revert CanNotWithdrawToZeroAddress();
+        uint256 currentBalance = paymasterIdBalances[msg.sender];
+        if (amount > currentBalance)
+            revert InsufficientBalance(amount, currentBalance);
+        paymasterIdBalances[msg.sender] =
+            paymasterIdBalances[msg.sender] -
+            amount;
```

Comment:
`paymasterIdBalances[msg.sender] = currentBalance - amount;` would be more optimized saves around 200gas

Reply:
done